### PR TITLE
Added the ability to fetch an animation from a url

### LINKED ISSUE #869: Bug: in docs, it says the SDK can "Download animations from the network"
This:

http://airbnb.io/lottie/android/avd.html

image

This is incorrect, as is shown here:
https://github.com/airbnb/lottie-android/issues/338

You can't set a url just in a single line. Some work need to be done first.

### PR DESCRIPTION
**Pull request by gpeal**
Fixes #869

### CODE REVIEW

**Comment by gpeal** [comment]
Snapshot Test Results: https://happo.io/compare?q=lottie-master-android26..lottie-63669d0b6c546c9200fd41dee60c375bcdfab975-android26&a=27

**Comment by gpeal** [comment]
Snapshot Test Results: https://happo.io/compare?q=lottie-master-android26..lottie-1d0a25e2a6d8ff02eb096b9ba4d6bbd2d61a07b6-android26&a=27

**Comment by gpeal** [comment]
@felipecsl @elihart @BenSchwab Could one of you take a look at this please?

File: lottie/src/main/java/com/airbnb/lottie/LottieCompositionFactory.java [hunk 1]
```diff
@@ -34,6 +35,32 @@
   private LottieCompositionFactory() {
   }
 
+  /**
+   * Fetch an animation from an http url. Once it is downloaded once, Lottie will cache the file to disk for
+   * future use. Because of this, you may call `fromUrl` ahead of time to warm the cache if you think you
```
Comment (elihart) [inline lottie/src/main/java/com/airbnb/lottie/LottieCompositionFactory.java]: caching++

File: lottie/src/main/java/com/airbnb/lottie/LottieCompositionFactory.java [hunk 2]
```diff
@@ -63,6 +91,11 @@ private LottieCompositionFactory() {
     }
   }
 
+  /**
+   * Parse an animation from raw/res. This is recommended over putting your animation in assets because
```
Comment (elihart) [inline lottie/src/main/java/com/airbnb/lottie/LottieCompositionFactory.java]: nice doc improvements 👍

File: lottie/src/main/java/com/airbnb/lottie/network/NetworkFetcher.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    } else {
+      result = LottieCompositionFactory.fromJsonInputStreamSync(inputStream, url);
+    }
+    if (result.getValue() != null) {
+      return result.getValue();
+    }
+    return null;
+  }
+
+  private LottieResult<LottieComposition> fetchFromNetwork() {
```
Comment (elihart) [inline lottie/src/main/java/com/airbnb/lottie/network/NetworkFetcher.java]: @WorkerThread annotations on these?

File: lottie/src/main/java/com/airbnb/lottie/LottieTask.java [hunk 4]
```diff
@@ -23,6 +26,8 @@
  */
 public class LottieTask<T> {
 
+  private static final Executor EXECUTOR = Executors.newCachedThreadPool();
```
Comment (elihart) [inline lottie/src/main/java/com/airbnb/lottie/LottieTask.java]: This seems like something people might want to customize. The easiest way to enable that would be to make it public and non final
that's an approach i started using to set epoxy defaults with the least amount of boilerplate

**Comment by gpeal** [comment]
Snapshot Test Results: https://happo.io/compare?q=lottie-master-android26..lottie-ef5b123d7982bc16874bb9d87d4634830ca47e6f-android26&a=27

**Comment by gpeal** [comment]
Snapshot Test Results: https://happo.io/compare?q=lottie-master-android26..lottie-d582b3b13bab02b2e47a5aaa768536081a04563f-android26&a=27

**Comment by gpeal** [comment]
Snapshot Test Results: https://happo.io/compare?q=lottie-master-android26..lottie-207071545004b3bbc8360e6f9ce621596e876501-android26&a=27

**Pull request merged**